### PR TITLE
Add picard tool?

### PR DESCRIPTION
**Pull request by aerijman**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by bwlang — CHANGES_REQUESTED** [review]
looks good - just needs some tests

File: tools/picard/picard_macros.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
     ]]></token>
 
     <token name="@java_options@"><![CDATA[
-    _JAVA_OPTIONS=\${_JAVA_OPTIONS:-'-Xmx2048m -Xms256m'} &&
+    _JAVA_OPTIONS=\${_JAVA_OPTIONS:-'-Xmx4084m -Xms256m'} &&
```
Comment (bwlang) [inline tools/picard/picard_macros.xml]: does this really need 4 G to run?
that's ok for our cluster but might be a problem for others.

File: tools/picard/picard_CollectSequencingArtifactsMetrics.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            <validator type="no_options" message="No indexes are available" />
+          </options>
+          <validator type="no_options" message="A built-in dictionary is not available for the build associated with the selected input file"/>
+        </param>
+      </when>
+      <when value="history">
+        <param name="ref_file" type="data" format="fasta" label="Use the following dataset to create dictionary" help="You can upload a FASTA sequence to the history from which Picard will automatically generate dictionary using CreateSequenceDictionary command" />
+      </when>
+    </conditional>
+    <!--<param name='reference_fasta_filename' type='data' label='Reference genome' help='last reference genome used is default'/>-->
```
Comment (bwlang) [inline tools/picard/picard_CollectSequencingArtifactsMetrics.xml]: delete?

**Review by bwlang — APPROVED** [review]
Just a couple of comments...  once those are resolved, I think this looks ready for upstream.
You should probably rebase on their master before submitting the PR.

**Comment by aerijman** [comment]
Thank you!

I added it as a priority to my tomorrow’s morning calendar.

Ariel

File: tools/picard/picard_macros.xml (hunk 1 above)
Comment (aerijman) [reply tools/picard/picard_macros.xml]: It works with 2G. In new commit is going to be 2G.

File: tools/picard/picard_CollectSequencingArtifactsMetrics.xml (hunk 2 above)
Comment (aerijman) [reply tools/picard/picard_CollectSequencingArtifactsMetrics.xml]: Should I uncomment SPNs? 
I didn't test the SNPs yet. should I?

Comment (bwlang) [reply tools/picard/picard_CollectSequencingArtifactsMetrics.xml]: Isn't that commented line repalced by the preceeding lines?
maybe I'm missing something.

Comment (aerijman) [reply tools/picard/picard_CollectSequencingArtifactsMetrics.xml]: Yes. To be sure I tested it after deleting the line and it works.
I also have commented the line responsible to allow to input a vcf file and I was wondering if I should leave it commented, exclude it or uncomment it.

Comment (bwlang) [reply tools/picard/picard_CollectSequencingArtifactsMetrics.xml]: I think we should dump the vcf line for now I don't see much value in it, can always add it later if needed

**Comment by aerijman** [comment]
Is it ready to go now?

I accept guidance in the pull request to galaxyproject...

**Comment by bwlang** [comment]
Yeah.. go ahead and send it upstream!

Brad

On Apr 24, 2019, at 10:45 AM, Ariel Erijman <notifications@github.com<mailto:notifications@github.com>> wrote:


Is it ready to go now?

I accept guidance in the pull request to galaxyproject...

**Pull request merged**